rest_framework/decorators.py

```diff
@@ -130,29 +130,45 @@ def decorator(func):
     return decorator
 
 
-def detail_route(methods=None, **kwargs):
+def action(methods=None, detail=True, url_path=None, url_name=None, **kwargs):
```

Comment:
So, the `detail` parameter...:

* Do we want an implied default? Or do we want to make it required, with a helpful assertion? 
* If implied which is the best default? i.e. Do we generate `list` or `detail` routes if you don't pass the parameter. 
* Do we want to document whichever we choose as not passing the parameter? `@action(detail=True)` and `@action()` in all the examples, say? 

Not sure what I think there.

Reply:
> Do we want an implied default? Or do we want to make it required, with a helpful assertion?

I don't think a default is beneficial here. Making it a required arg seems 👍 to me.